### PR TITLE
added a trycatch for api reserve_edi_id

### PR DESCRIPTION
**Pull request by InigoP**

### CODE REVIEW

File: R/api-functions.R [hunk 1]
```diff
@@ -54,50 +66,61 @@ reserve_edi_id <- function(user_id, password, environment = "production") {
...
-      }
-      if (max_iter > iter) {
-        print("Request timed out, check that you inputs are all valid and try again")
-        break 
-      }
+    response <- httr::GET(
+      url = paste0(base_url, "evaluate/report/eml/", transaction_id),
+      config = httr::authenticate(
+        paste0('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'),
+        password,
```
Comment (alvizek) [inline R/api-functions.R:99]:
```suggestion
        password
```

There is an extra comma that was throwing an error

File: R/api-functions.R [hunk 2]
```diff
@@ -121,8 +144,13 @@ evaluate_edi_package <- function(user_id, password, eml_file_path, environment =
 #'                             eml_file_path = "data/edi20.1.xml")}
 #' @export
```
Comment (alvizek) [inline R/api-functions.R:146]: I did some testing using the api-update branch. I was able to upload 1239.1 to the staging environment but the `upload_edi_package` returned the following message: `No encoding supplied: defaulting to UTF-8.
Error in EMLaide::upload_edi_package(user_id = Sys.getenv("EDI_USER_ID"),  : 
  RequestErrorRequest failed, check that you inputs are all valid and try again. 
          See more information on request status below: create_168550079871746301`

I checked on the staging web portal to see if it uploaded even though I got this error and saw that it was uploaded.

**Review by alvizek — CHANGES_REQUESTED** [review]
I did some testing and noted two issues. After these are resolved I can do additional more thorough testing.

File: R/api-functions.R (hunk 2 above)
Comment (ErinCain) [reply R/api-functions.R:146]: I am also getting this errror for both upload and evaluating.

**Comment by ErinCain** [comment]
I am not seeing a big change in timing based on what I set maxiter to. Lets check to make sure this is doing what we want it to do.

**Pull request closed without merging**